genestack_client/data_importer.py

```diff
@@ -124,6 +125,40 @@ def __invoke_loader(self, parent, importer_type, metainfo):
         self.__process_links(metainfo)
         return self.importer.invoke('createFile', parent, importer_type, metainfo)['accession']
 
+    @staticmethod
+    def __add_to_metainfo(metainfo, key, value, value_type, required=False):
+        """
+        Add value to metainfo, check if value is not already present,
+        checks required fields.
+
+        :param metainfo: metainfo object to be updated
+        :type metainfo: Metainfo
+        :param key: key to add
+        :type key: str
+        :param value: single argument to pass to the value class constructor, or list if need to add many values.
+        :type value: object
+        :param value_type: value class
+        :type value_type: T => genestack_client.metainfo_scalar_values.MetainfoScalarValue
+        :param required: flag if value is required
+        :type required: bool
+        :return: None
+        """
+        current_value = metainfo.get(key)
+        if current_value is not None and value is not None:
+            raise GenestackException('Key "%s" is passed both as function argument '
+                                     'and inside metainfo object' % key)
+        if current_value:
+            return
+
+        if required and value is None:
+                raise GenestackException('Missing required key "%s", '
+                                         'it should be passed as function argument '
+                                         'or in metainfo object' % key)
+        if value is not None:
```

Comment:
But when `required == False` and `value is None` we don't add `key: None` to metainfo.
Is it expected behaviour?  Feels wrong to me, but if it is, it warrants a comment.